lib/saluki-env/src/autodiscovery/mod.rs

```diff
@@ -162,16 +208,35 @@ fn bytes_to_hashmap(bytes: Vec<u8>) -> Result<HashMap<MetaString, serde_yaml::Va
     Ok(result)
 }
 
+fn map_to_bytes(map: &HashMap<MetaString, serde_yaml::Value>) -> Result<Vec<u8>, serde_yaml::Error> {
+    let s = serde_yaml::to_string(map)?;
+    Ok(s.into_bytes())
+}
+
+fn get_name_for_instance(instance: &HashMap<MetaString, serde_yaml::Value>) -> String {
+    if let Some(name) = instance.get("name") {
+        if let Some(value) = name.as_str() {
+            return value.to_string();
+        }
+    }
+    if let Some(namespace) = instance.get("namespace") {
+        if let Some(value) = namespace.as_str() {
+            return value.to_string();
+        }
+    }
+    "".to_string()
+}
```

Comment:
Functions like this are another good reason why a check config instance could stand to be wrapped up in a custom struct, to hide this logic and simply expose it as a basic method.